crates/types/src/sessionid.rs

```diff
@@ -0,0 +1,62 @@
+use core::fmt;
+
+use bitcoin::{
+    consensus::{encode::Error, Decodable, Encodable},
+    hashes::{sha256, Hash},
+    io::Cursor,
+};
+use serde::{Deserialize, Serialize};
+
+/// A unique identifier of signatures and nonces exchange session made by
+/// hashing unique data related to it.
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Debug, Deserialize, Serialize)]
+pub struct SessionId(sha256::Hash);
+
+impl SessionId {
+    pub const fn hash(data: &[u8]) -> Self {
+        Self(sha256::Hash::const_hash(data))
+    }
+
+    pub fn from_bytes(bytes: &[u8]) -> Result<Self, Error> {
+        let mut cursor = Cursor::new(bytes);
+        let session_id = Decodable::consensus_decode(&mut cursor)?;
+        Ok(Self(session_id))
+    }
```

Comment:
Should specify in the fn name that this is specifically about a consensus-encodable type.

But I'm not sure what purpose this would ever have?

Reply:
I'll get rid of it then. Just thought, that reusing implementation of decoding and encoding for types from Bitcoin is easier